internal/impl/kafka/lag.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafka
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/redpanda-data/benthos/v4/public/service"
+	"github.com/redpanda-data/connect/v4/internal/asyncroutine"
+	"github.com/twmb/franz-go/pkg/kadm"
+	"github.com/twmb/franz-go/pkg/kgo"
+)
+
+// ConsumerLag is a struct that manages the consumer lag for Kafka topics.
+type ConsumerLag struct {
+	lagUpdater    *asyncroutine.Periodic
+	topicLagCache *sync.Map
+}
+
+// NewConsumerLag creates a new ConsumerLag instance.
+func NewConsumerLag(
+	client *kgo.Client,
+	consumerGroup string,
+	logger *service.Logger,
+	topicLagGauge *service.MetricGauge,
+	topicLagRefreshPeriod time.Duration,
+) *ConsumerLag {
+	adminClient := kadm.NewClient(client)
+	topicLagCache := &sync.Map{}
```

Comment:
How about new(sync.Map)

Reply:
I tend to use this syntax - I'm not sure if there is a difference either way, but happy to switch